### PR TITLE
chore: rely less on the index.from_src pattern

### PR DESCRIPTION
**Pull request by alexeagle**
Instead use substitutions when building packages to switch out the local bin label with the npm one

### CODE REVIEW

File: packages/typescript/src/internal/build_defs.bzl [hunk 1]
```diff
@@ -22,7 +22,8 @@ load("@build_bazel_rules_typescript//internal:common/compilation.bzl", "COMMON_A
 load("@build_bazel_rules_typescript//internal:common/tsconfig.bzl", "create_tsconfig")
 load("//internal:ts_config.bzl", "TsConfigInfo")
 
-_DEFAULT_COMPILER = "@npm//@bazel/typescript/bin:tsc_wrapped"
+# NB: substituted with "@npm//@bazel/typescript/bin:tsc_wrapped" in the pkg_npm rule
+_DEFAULT_COMPILER = "@build_bazel_rules_typescript//internal:tsc_wrapped_bin"
```
Comment (jbedard) [inline packages/typescript/src/internal/build_defs.bzl]: Does `@build_bazel_rules_typescript` work different then the other `@npm_bazel_*` targets? So we can't stick with the `@npm//@bazel/X` => `@npm_bazel_X` convention here?

**Comment by jbedard** [comment]
If we do this in a limited and consistent fashion I think it might be better then all the `from_src` imports. Knowing it will be substituted while reading the code is important, but might be better/no-worse then needing to know the rule might be executed from `from_src.bzl`?

File: packages/typescript/src/internal/build_defs.bzl (hunk 1 above)
Comment (alexeagle) [reply packages/typescript/src/internal/build_defs.bzl]: yeah it's a weird snowflake because rules_typescript has a g3 source-of-truth

**Comment by alexeagle** [comment]
@jbedard agreed, we need the repo to be more contributor friendly so your take on it has more weight than the rocket science Greg or I might try to apply to it.

I can spend some more time to kill the other .from_src usages but they're all less trivial than this pattern.

Also this will help us down the road if we have a BUILD file generator, we'll be able to use it ourselves in these cases since the load statement looks like what we tell users.

Note, we plan at some point to kill the `@npm_something` workspaces since these should be load from `@npm//something` just like the generated npm_package_bin rules for non-bazel-aware packages

**Pull request merged**